pycaret/internal/pycaret_experiment/supervised_experiment.py

```diff
@@ -1111,20 +1112,24 @@ def _create_model_with_cv(
             fit_kwargs = get_pipeline_fit_kwargs(pipeline_with_model, fit_kwargs)
             self.logger.info(f"Cross validating with {cv}, n_jobs={n_jobs}")
 
-            model_fit_start = time.time()
-            with redirect_output(self.logger):
-                scores = cross_validate(
-                    pipeline_with_model,
-                    data_X,
-                    data_y,
-                    cv=cv,
-                    groups=groups,
-                    scoring=metrics_dict,
-                    fit_params=fit_kwargs,
-                    n_jobs=n_jobs,
-                    return_train_score=return_train_score,
-                    error_score=0,
-                )
+            # Monkey patch sklearn's _fit_and_score function to allow
+            # for pipelines that drop samples during transformation
+            with patch("sklearn.model_selection._validation._fit_and_score", fs):
```

Comment:
does this actually work with joblib (n_jobs>1)? I think it may not propagate correctly to processes spawned by joblib.

Reply:
It does. I tested it. The way it works is: the `_fit_and_score` function is the one that is propagated to the workers. This function is patched, and in turn, patches the `_score` function, which is the actual function that we want patched.